### PR TITLE
Make sure computable pointers don't appear in dump

### PR DESCRIPTION
**Pull request by elprans**
The dump/restore implementation currently erroneously considers
computable properties and links as real and includes them in the dump.
This slipped through testing because of another bug, which causes
columns to be generated for `single` properties and links. `multi`
computables, however, are broken and it is impossible to generate a dump
of a schema containing them.

Fix this by properly excluding non-dumpable pointers.  Unfortunately,
this bug causes previously-generated dumps to contain extraneous (empty)
columns, so this patch includes a procedure to rewrite the binary `COPY`
stream to eject those columns.

A portion of this patch will be backported to a7 and perhaps earlier to
make sure those versions can produce working dumps on schemas with
`multi` computables.

Issue: #2057

### CODE REVIEW

File: edb/server/compiler/compiler.py [hunk 1]
```diff
@@ -2139,6 +2139,11 @@ async def describe_database_restore(
         )
         ctx.state.start_tx()
 
+        dump_with_extraneous_computables = (
+            dump_server_ver is not None
+            and dump_server_ver < (1, 0, verutils.VersionStage.ALPHA, 8)
```
Comment (tailhook) [inline edb/server/compiler/compiler.py:2144]: Is it still fine to use on patched version of edgedb?

File: edb/server/compiler/compiler.py [hunk 2]
```diff
@@ -2139,6 +2139,11 @@ async def describe_database_restore(
         )
         ctx.state.start_tx()
 
+        dump_with_extraneous_computables = (
+            dump_server_ver is not None
```
Comment (tailhook) [inline edb/server/compiler/compiler.py:2143]: Why version could be `None`? I would say it's better to assert on it, this condition sounds like we have "always latest version" dumps, which is probably nonsensical.

**Review by tailhook — APPROVED** [review]
Other than two questions, looks good to me.

File: edb/server/pgcon/pgcon.pyx [hunk 3]
```diff
@@ -947,10 +956,90 @@ cdef class PGProto:
...
+
+        real_ncols = ncols + len(elided_cols)
+        frb_init(&rbuf, data, data_len)
+
+        elide = <int8_t*>cpythonx.PyMem_Calloc(
+            <size_t>real_ncols, sizeof(int8_t))
+
+        try:
+            for col in elided_cols:
+                elide[col] = 1
```
Comment (1st1) [inline edb/server/pgcon/pgcon.pyx:989]: This calloc and the initialization of `elide` can me be moved outside of this function to `async def _restore`

**Review by 1st1 — APPROVED** [review]

File: edb/server/compiler/compiler.py (hunk 2 above)
Comment (1st1) [reply edb/server/compiler/compiler.py:2143]: I'm also curious why it can be `None` -- perhaps add a comment at least?

Comment (elprans) [reply edb/server/compiler/compiler.py:2143]: The earliest versions did not have the version header in dumps. I'll add a comment

File: edb/server/compiler/compiler.py (hunk 1 above)
Comment (elprans) [reply edb/server/compiler/compiler.py]: Yes. The backpatched version only takes care of the `multi` computable case, leaving the old behavior of dumping `single` computable columns in place.

File: edb/server/compiler/compiler.py (hunk 2 above)
Comment (tailhook) [reply edb/server/compiler/compiler.py]: Then `None` should be treated as a smaller version not larger. I.e. `dump_with_extraneous_computables` should be `True` for `None` not `False`.

Comment (tailhook) [reply edb/server/compiler/compiler.py]: Ah, I see you fixed this.

**Pull request merged**